### ISSUE TITLE
Local variables with items

### ISSUE THREAD

**Issue by rokob** [rollbar/rollbar-php#173]
We capture all local variables in Python has an option as part of our SDK. It would be nice to have that option in PHP. http://stackoverflow.com/questions/5101942/dump-all-php-variables might be a starting point along with https://github.com/rollbar/pyrollbar/blob/97623876abeb200182bcc98b4f598dd93f9efcfe/rollbar/__init__.py#L769

**Comment by ArturMoczulski** [comment]
I don't see how this would be possible in PHP. There is nothing like inspect module and get_defined_vars() in PHP only provides information about variables within local scope. From extensive research and my own experimentation I don't see any way of providing locals dump for all stack frames.

**Comment by brianr** [comment]
@ArturMoczulski Thanks for researching this - that's too bad. However, looks like we can at least get values of function arguments. How about if we change the scope of this story to that?

Example showing values of function arguments being available:

```php
function greet($name) {
    say("Hello, " . $name);

}

function say($phrase) {
    throw new Exception($phrase);
}

try {
    greet("Artur");
} catch (Exception $e) {
    print_r($e);
}
```

Output:

```
Exception Object
(
    [message:protected] => Hello, Artur
    [string:Exception:private] => 
    [code:protected] => 0
    [file:protected] => /Users/brian/Development/phptest/test.php
    [line:protected] => 19
    [trace:Exception:private] => Array
        (
            [0] => Array
                (
                    [file] => /Users/brian/Development/phptest/test.php
                    [line] => 14
                    [function] => say
                    [args] => Array
                        (
                            [0] => Hello, Artur
                        )

                )

            [1] => Array
                (
                    [file] => /Users/brian/Development/phptest/test.php
                    [line] => 23
                    [function] => greet
                    [args] => Array
                        (
                            [0] => Artur
                        )

                )

        )

    [previous:Exception:private] => 
)
```

**Comment by ArturMoczulski** [comment]
@brianr Yeah, but this is already available as part of the exception trace and message trace implemented in 1.1.0

This is a partial dump of information that is already sent to Rollbar from 1.1.0 with traces
```json
"backtrace": [
        {
          "function": "getMessage", 
          "args": [
            "Testing PHP Notifier", 
            []
          ], 
          "object": {}, 
          "file": "/home/ubuntu/workspace/src/DataBuilder.php", 
          "line": 326, 
          "type": "->", 
          "class": "Rollbar\\DataBuilder"
        }, 
        {
          "function": "getBody", 
          "args": [
            "Testing PHP Notifier", 
            []
          ], 
          "object": {}, 
          "file": "/home/ubuntu/workspace/src/DataBuilder.php", 
          "line": 293, 
          "type": "->", 
          "class": "Rollbar\\DataBuilder"
        }, 
        {
          "function": "makeData", 
          "args": [
            "warning", 
            "Testing PHP Notifier", 
            []
          ], 
          "object": {}, 
          "file": "/home/ubuntu/workspace/src/Config.php", 
          "line": 275, 
          "type": "->", 
          "class": "Rollbar\\DataBuilder"
        }, 
        {
          "function": "getRollbarData", 
          "args": [
            "warning", 
            "Testing PHP Notifier", 
            []
          ], 
          "object": {}, 
          "file": "/home/ubuntu/workspace/src/RollbarLogger.php", 
          "line": 52, 
          "type": "->", 
          "class": "Rollbar\\Config"
        }, 
        {
          "function": "getPayload", 
          "args": [
            "eb2561a52efb4d4bba5a1d4b68be13e9", 
            "warning", 
            "Testing PHP Notifier", 
            []
          ], 
          "object": {}, 
          "file": "/home/ubuntu/workspace/src/RollbarLogger.php", 
          "line": 37, 
          "type": "->", 
          "class": "Rollbar\\RollbarLogger"
        }, 
        {
          "function": "log", 
          "args": [
            "warning", 
            "Testing PHP Notifier", 
            []
          ], 
          "object": {}, 
          "file": "/home/ubuntu/workspace/tests/RollbarLoggerTest.php", 
          "line": 32, 
          "type": "->", 
          "class": "Rollbar\\RollbarLogger"
        }, 
        {
          "function": "testLog", 
          "object": {}, 
          "type": "->", 
          "class": "Rollbar\\RollbarLoggerTest", 
          "args": []
        }
...
}
```

**Comment by brianr** [comment]
Hm, doesn't seem to be working. I tried this a few minutes ago (before posting my previous comment) using 1.1.1, and there are no stack trace variable values. Example: https://rollbar.com/Rollbar/rollbar-php/items/111/occurrences/25275004700/

**Comment by ArturMoczulski** [comment]
I see. What happened there. Here is a PR that adds stack frames arguments values to the traces: https://github.com/rollbar/rollbar-php/pull/178

**Comment by rokob** [comment]
Merged with #192
